Description: Better toolbar layout
 Better toolbar layout - use separators and search button at end of toolbar
 .
 nautilus (1:3.4.2-2solusos3) two; urgency=low
 .
   * Better toolbar layout
Author: Ikey Doherty <ikey@solusos.com>

---
Reviewed-By: Ikey Doherty <ikey@solusos.com>
Last-Update: <2012-08-09>

--- nautilus-3.4.2.orig/src/nautilus-toolbar-ui.xml
+++ nautilus-3.4.2/src/nautilus-toolbar-ui.xml
@@ -4,9 +4,12 @@
 	<toolitem name="Forward" action="Forward"/>
         <toolitem name="Up" action="Up"/>
  	<toolitem name="Reload" action="Reload"/>
+	<separator name="AfterNavigation" />
         <toolitem name="Edit" action="Edit"/>
         <toolitem name="Home" action="Home"/>
         <toolitem name="Computer" action="Computer"/>
+	<placeholder />
+	<separator name="BeforeSearch" />
         <toolitem name="Search" action="Search"/>
 </toolbar>
 </ui>
--- nautilus-3.4.2.orig/src/nautilus-toolbar.c
+++ nautilus-3.4.2/src/nautilus-toolbar.c
@@ -124,6 +124,8 @@ nautilus_toolbar_constructed (GObject *o
 	GtkWidget *hbox, *toolbar, *search;
 	GtkStyleContext *context;
 
+	GtkWidget *sep_space;
+
 	G_OBJECT_CLASS (nautilus_toolbar_parent_class)->constructed (obj);
 
 	gtk_style_context_set_junction_sides (gtk_widget_get_style_context (GTK_WIDGET (self)),
@@ -142,6 +144,10 @@ nautilus_toolbar_constructed (GObject *o
 	context = gtk_widget_get_style_context (toolbar);
 	gtk_style_context_add_class (context, GTK_STYLE_CLASS_PRIMARY_TOOLBAR);
 
+	sep_space = gtk_ui_manager_get_widget(self->priv->ui_manager, "/Toolbar/BeforeSearch");
+	gtk_separator_tool_item_set_draw (GTK_SEPARATOR_TOOL_ITEM (sep_space), FALSE);
+	gtk_tool_item_set_expand (GTK_SEPARATOR_TOOL_ITEM (sep_space), TRUE);
+
 	search = gtk_ui_manager_get_widget (self->priv->ui_manager, "/Toolbar/Search");
 	gtk_style_context_add_class (gtk_widget_get_style_context (search), GTK_STYLE_CLASS_RAISED);
 	gtk_widget_set_name (search, "nautilus-search-button");
@@ -164,7 +170,7 @@ nautilus_toolbar_constructed (GObject *o
 	gtk_tool_item_set_expand (item, TRUE);
 	gtk_container_add (GTK_CONTAINER (item), hbox);
 	/* append to the end of the toolbar so navigation buttons are at the beginning */
-	gtk_toolbar_insert (GTK_TOOLBAR (self->priv->toolbar), item, -1);
+	gtk_toolbar_insert (GTK_TOOLBAR (self->priv->toolbar), item, 8);
 	gtk_widget_show (GTK_WIDGET (item));
 
 	/* search bar */
